packages/protocol/contracts/Median.sol

```diff
@@ -0,0 +1,441 @@
+   * @param left the left index to begin search.
+   * @param right the right index to begin search.
+   * @param k
+   * @param selectKplusOne a bool representing whether the function should
+   *                       return the (k+1)st element or not.
+   * @return a tuple (i, j) where i and j are the indices of the kth and
+   *         (k+1)st elements of `arr`, respectively. In the case where
+   *         `selectKplusOne` is false, the tuple returned is (i, 0).
+   */
+   function _quickSelect
+  (
+    int256[] memory arr,
+    uint256 left,
+    uint256 right,
+    uint256 k,
+    bool selectKplusOne
+  )
+    private
+    pure
+    returns (uint256, uint256)
+  {
+    assert(left <= k && k <= right);
+    
+    if (left == right) {return (k, 0);}
+   
+    uint256 pivotIdx = partition(arr, left, right);
+
+    uint256 idx1;
+    uint256 idx2;
+
+    if (k == pivotIdx) {
+      idx1 = pivotIdx;
+    } else if (k < pivotIdx){
+      (idx1, idx2) = _quickSelect(arr, left, pivotIdx - 1, k, false);
+    } else {
+      (idx1, idx2) = _quickSelect(arr, pivotIdx + 1, right, k, false);
+    }
+    if (!selectKplusOne) {
+      return (idx1, 0);
+    }
+    assert(idx1 != arr.length - 1);
+    // In order to find (k+1)th element,
+    // find minimum in right partition of array
+    idx2 = idx1 + 1;
+    for (uint i=idx1+2; i<arr.length; i++) {
+      if (arr[i] < arr[idx2]) {
+        idx2 = i;
+      }
+    }
+    return (idx1, idx2);
+  }
+
+  /**
+   * Partitions the array in-place using a modified Hoare's partitioning
+   * scheme. Only elements between indices `lo` and `high` (inclusive) will be
+   * partitioned.
+   * @dev Hoare's algorithm is modified in order to return the index of the
+   *      pivot element.
+   * @return the index of the pivot
+   */
+  function partition
+  (
+    int256[] memory arr,
+    uint256 lo,
+    uint256 hi
+  )
+    private
+    pure
+    returns (uint256)
+  {
+    if (lo == hi) {return lo;}
+    
+    // make middle element pivot
+    uint m = arr.length / 2;
+    (arr[m], arr[lo] = arr[lo], arr[m]);
+    int pivot = arr[lo];
+
+    uint i = lo;
+    uint j = hi + 1;
+ 
+    while (true) {
+      do {
+        i++;
+      } while (i < arr.length && arr[i] < pivot);
+      do {
+        j--;
+      } while (arr[j] > pivot);
+      if (i >= j) {
+        // swap with pivot
+        (arr[lo], arr[j]) = (arr[j], arr[lo]);
+        return j;
+      } 
+      (arr[i], arr[j]) = (arr[j], arr[i]);
+    }
+  }
+
+  /* Return the kth element of a small array (at most length 16).
+   * @param arr an array of signed integers (at most length 16)
+   * @return the kth smallest element in `arr`
+   */ 
+  function _selectK_smallArray
+  (
+    int256[] memory arr,
+    uint256 k
+  )
+      private
+      pure
+      returns (int256)
+  {
+    assert(arr.length <= SMALL_ARRAY_MAX_LENGTH);
+    assert(0 <= k && k <= arr.length - 1);
+
+    if (arr.length == 1) {
+    }
+    else if (arr.length == 2) {
```

Comment:
```suggestion
    // if (arr.length == 1) {} else
    if (arr.length == 2) {
```

Reply:
If you want to be explicit about it

Reply:
Then `arr.length == 1` falls under the `else` case. So it's a matter of having a `arr.length == 1` check or having an `arr.length == 16` check. And, on second thought, I think it might be better to do this check in the topmost functions, and get it "out of the way" immediately.

Reply:
Ah, right sorry. It's just that the linter complains about empty blocks, so maybe replace `else` with `else if (arr.length == 16)`